pkg/azureupdate/common.go

```diff
@@ -68,6 +69,11 @@ func upgradeAllowed(ctx context.Context, g8sclient versioned.Interface, oldVersi
 			return false, microerror.Maskf(invalidOperationError, "downgrading is not allowed (attempted to downgrade from %s to %s)", oldVersion, newVersion)
 		}
 
+		// Check if either version is an alpha one.
+		if isAlphaRelease(oldVersion.String()) || isAlphaRelease(newVersion.String()) {
```

Comment:
Would it make any sense to block the upgrade if release name isn't plain semver `XX.YY.ZZ`? For now we don't have RCs or anything similar where this upgrade would be useful.

Reply:
Not sure about this.
I can imagine a future where we do beta releases that have upgrades allowed.
Plus we sometimes use named releases such as `13.0.0-yulia` and we might want to test upgrades from and to those releases.

Reply:
Ack. I'm fine with that.